pkgs/build-support/rust/build-rust-package/default.nix

```
@@ -97,6 +99,8 @@ assert useSysroot -> !(args.doCheck or true);
 
 stdenv.mkDerivation ((removeAttrs args [ "depsExtraArgs" "cargoUpdateHook" "cargoLock" ]) // lib.optionalAttrs useSysroot {
   RUSTFLAGS = "--sysroot ${sysroot} " + (args.RUSTFLAGS or "");
+} // lib.optionalAttrs isDarwinDebug {
+  RUSTFLAGS = "-C split-debuginfo=packed " + lib.optionalString useSysroot "--sysroot ${sysroot} " + (args.RUSTFLAGS or "");
```

```suggestion
  RUSTFLAGS = "-C split-debuginfo=packed"
    + lib.optionalString useSysroot " --sysroot ${sysroot} "
    + (args.RUSTFLAGS or "");
```
we don't want to add a trailing space here

---

As a matter of fact, this change makes my test case fail: https://github.com/n8henrie/nix-rust-debug-info

It fails because I've set `RUSTFLAGS = "-Cpanic=abort ";`: https://github.com/n8henrie/nix-rust-debug-info/blob/c1bef6c8aec1495b213205e49a788ed0c02bb1eb/default.nix#L19

which results in https://github.com/n8henrie/nix-rust-debug-info/actions/runs/10509248391/job/29114981376#step:5:3002

```
error: incorrect value `packed-Cpanic=abort` for codegen option `split-debuginfo` - one of supported split-debuginfo modes (`off`, `packed`, or `unpacked`) was expected
```

It seems to me like leaving a trailing space on all of these (so that user provided values don't get clobbered) was the right way to go, but I suppose something like this might be an (uglier) alternative?

```
lib.concatStringSep " " [ "-C split-debuginfo=packed" ] ++
    lib.optionals useSysroot ["--sysroot ${sysroot}"] ++
    [args.RUSTFLAGS or ""];
```

---

Aside from aesthetics, I don't see a reason to do remove the whitespaces and as @n8henrie explained this change will mess up the flags.

However, we can still write them over multiple lines if you think that looks better:

```nix
    RUSTFLAGS =
      "-C split-debuginfo=packed "
      + lib.optionalString useSysroot "--sysroot ${sysroot} "
      + (args.RUSTFLAGS or "");
```